File: ProyectoAsisTrack/src/main/java/com/example/ProyectoAsisTrack/Repository/DocentesRepository.java
package com.example.ProyectoAsisTrack.Repository;

import com.example.ProyectoAsisTrack.Model.Docentes;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface DocentesRepository extends JpaRepository<Docentes, Long> {
}

File: app/src/main/java/com/example/proyectoasistrack/data/Repository/DocentesRepository.kt
package com.example.proyectoasistrack.data.repository

import com.example.proyectoasistrack.data.dao.DocenteDao
import com.example.proyectoasistrack.data.model.Docente
import kotlinx.coroutines.flow.Flow

class DocentesRepository(private val docenteDao: DocenteDao) {

    fun getAllDocentes(): Flow<List<Docente>> = docenteDao.getAll()
    suspend fun insertDocente(docente: Docente) = docenteDao.insert(docente)
    suspend fun updateDocente(docente: Docente) = docenteDao.update(docente)
    suspend fun deleteDocente(docente: Docente) = docenteDao.delete(docente)
}